### PR TITLE
Opi hackathon block server

### PR DESCRIPTION
**Pull request by AdrianPotter**
ISISComputingGroup/IBEX/issues/1129

### CODE REVIEW

**Comment by mattclarke** [comment]
This is fundamentally okay, but the file watcher gets upset about the new screens.xml file.
I don't think this is an easy fix.

File: BlockServer/core/devices_manager.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+import os
+from server_common.utilities import print_and_log, compress_and_hex
+from BlockServer.core.file_path_manager import FILEPATH_MANAGER
+from BlockServer.fileIO.schema_checker import ConfigurationSchemaChecker
+from xml.dom import minidom
+
+GET_SCREENS = "GET_SCREENS"
+SET_SCREENS = "SET_SCREENS"
+
+SCREENS_FILE = "screens.xml"
```
Comment (John-Holt-Tessella) [inline BlockServer/core/devices_manager.py]: Does this not repeat FILENAME_SCREENS in constants.py

File: BlockServer/core/devices_manager.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        self._schema_folder = schema_folder
+        self._cas = cas
+        self._devices_pvs = dict()
+        self._vc = vc_manager
+        self._bs = block_server
+        self._current_config = ""
+
+    def load_current(self):
+        """Create the PVs for all the devices found in the devices directory."""
+        devices_file_name = self.get_devices_filename()
```
Comment (John-Holt-Tessella) [inline BlockServer/core/devices_manager.py]: I think that if the file doesn't exist this returns blank and an exception will be thrown and caught about a file with no file name does not exist

File: BlockServer/fileIO/schema_checker.py [hunk 3]
```diff
@@ -104,22 +104,36 @@ def check_config_file_matches_schema(schema_folder, config_xml_path, is_componen
...
         folder, file_name = string.rsplit(schema_filepath, os.sep, 1)
         xmlparser = ConfigurationSchemaChecker._import_schema(folder, file_name)
 
         try:
-            etree.fromstring(synoptic_xml_data, xmlparser)
+            etree.fromstring(xml_data, xmlparser)
         except etree.XMLSyntaxError as err:
-            raise ConfigurationInvalidUnderSchema("Synoptic incorrectly formatted: " + str(err.message))
+            raise ConfigurationInvalidUnderSchema(str(err.message))
+
```
Comment (John-Holt-Tessella) [inline BlockServer/fileIO/schema_checker.py]: This is the same as  check_screens_match_schema generalise error message and combine

**Pull request merged**